jupyter-extension/jupyterlab_pachyderm/env.py

```diff
@@ -1,7 +1,11 @@
 import os
 from distutils.util import strtobool
 
+HTTP_UNIX_SOCKET_SCHEMA="http_unix"
+HTTP_SCHEMA="http"
 PFS_MOUNT_DIR = os.environ.get("PFS_MOUNT_DIR", "/pfs")
+PFS_SOCK_PATH = os.environ.get("PFS_SOCK_PATH", "/tmp/pfs.sock")
+DEFAULT_SCHEMA = os.environ.get("DEFAULT_SCHEMA", HTTP_UNIX_SOCKET_SCHEMA)
```

Comment:
The default schema should be HTTP_SCHEMA right?

Reply:
Ideal would be to have it use unix domain socket (HTTP_UNIX_SOCKET_SCHEMA), the only justification to use HTTP_SCHEMA would be for backward compatibility. Although I cannot see what difference it will make since for an enduser there should be no difference in behavior.

Reply:
AFAICT, `DEFAULT_SCHEMA` controls whether the JupyterLab backend connects to the mount server via a unix socket or via localhost networking (I am not sure why DEFAULT is in the name—there are no circumstances in which it would fall back from one to the other, are there? The value in DEFAULT_SCHEMA is the only connection scheme used?)

In any case, if this understanding is correct, I'd like to add a correction to the statement that "HTTP_SCHEMA would be for backward compatibility." The mount server often runs in a separate container from the JupyterLab backend (as a FUSE backend, when run in Kubernetes, it has to run in a privileged container. The JupyterLab backend does not need to run in a privileged container, and in fact can execute arbitrary user code and should therefore not be run in a privileged container if at all possible. Therefore a common configuration is to run the JupyterLab backend in an unprivileged container, and run the mount-server in a separate, privileged container). In this configuration, communication has to happen over a network interface.